AP_Mount: switch to RC_TARGETING on RC input

rmackay9 opened the pull request:
This is a slightly simplified version of PR https://github.com/ArduPilot/ardupilot/pull/25850

This PR makes it more convenient for users to retake manual control of the gimbal using RC input.  Currently users can only retake control after pushing a button on the ground station (see below).
![image](https://github.com/ArduPilot/ardupilot/assets/1498098/78781556-249c-4b40-89f2-b96d63887912)

With this new change, the Mount's mode is automatically changed to RC_TARGETING if the user moves one of the configured mount RC inputs by at least the channel's deadzone or 10pwm (whichever is larger).  The only exception is when the mode is in RETRACT mode.

This has been tested on real hardware (a CubeOrange paired with a Xacti camera gimbal) and I could easily move between controlling the gimbal with the RC inputs and Mission Planner's Payload screen.

IamPete1 reviewed (commented):
Should this check for RC failsafe with something like `rc().has_valid_input()`?

Do we need a mount option to allow turning this off?

rmackay9 commented:
@IamPete1,

I've added the valid RC check and re-tested successfully.

I suspect the check actually makes no difference because we already checked for rc input (on each channel) being non-zero but it can't hurt.

Also rebased on master.

IamPete1 reviewed (approved):
LGTM

rmackay9 commented:
Thanks for the approval! merged
